### DISCUSSION TITLE
C++ inference

### DISCUSSION THREAD

**Discussion by pfeatherstone** [google/flax#1643] [Q&A] [answered]
Is there a way to infer flax models in c++ ?

**Comment by jheek** [comment, marked as answer]
You can use [jax2tf](https://github.com/google/jax/tree/main/jax/experimental/jax2tf) to convert the model to a TF SavedModel which is then usable with all the standard tools TensorFlow provides for inference.

**Reply by pfeatherstone** [reply]
Hmm. Is `jax2onnx` on the roadmap?

**Reply by marcvanzee** [reply]
You could combine jax2tf and [tf2onnx](https://github.com/onnx/tensorflow-onnx) to get jax2onnx.

**Reply by pfeatherstone** [reply]
Ok cool. Though there is a very high chance that wouldn't work all the time. In my experience, chaining more than 1 model converter almost always fails. But thanks for pointing me to some repos.

**Reply by marcvanzee** [reply]
In our experience the jax2tf converter is very stable, and we've tried it on very large models. I can't say anything about the tf2onnx converter though.
